rq/worker.py

```diff
@@ -474,8 +474,17 @@ def perform_job(self, job):
         self.set_current_job_id(job.id)
         self.heartbeat((job.timeout or 180) + 60)
 
+        try:
+            func_name = job.func_name
+        except UnpickleError:
+            # Use the public setter here, to immediately update Redis
+            job.set_status(Status.FAILED)
+            self.log.exception('RQ failed to unpickle job %s' % job.id)
```

Comment:
In dealing with sentry, I've found the best solution is to do simple things like:

```python
except Exception as e:
    self.log.exception(e, extra={...})
```

With any got exception handler, passing the exception itself is the most supported approach, and sentry especially does a good job giving context and insight.

Basically you can reduce a ton of code and verbosity, as well as improve support.